imageio_dng.c error: ‘TIFFTAG_PREVIEWCOLORSPACE’ undeclared (first use in this function)

### Is there an existing issue for this?

- [x] I checked and did not find my issue in the already reported ones

### Describe the bug

when compiling:
```
....
/home/a/darktable/src/imageio/imageio_dng.c: In function ‘_register_extra_dng_fields’:
/home/a/darktable/src/imageio/imageio_dng.c:82:7: error: ‘TIFFTAG_PREVIEWCOLORSPACE’ undeclared (first use in this function)
   82 |     { TIFFTAG_PREVIEWCOLORSPACE, 1, 1, TIFF_LONG, FIELD_CUSTOM,
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~

```

### Steps to reproduce

 bash ../build.sh --prefix /opt/darktable --build-type Release --enable-ai

### Where did you obtain darktable from?

darktable.org / GitHub release

### darktable version

current master

### What OS are you using?

Linux

### What is the version of your OS?

debian 12

> /home/a/darktable/src/imageio/imageio_dng.c:82:7: error: ‘TIFFTAG_PREVIEWCOLORSPACE’ undeclared (first use in this function)

TIFFTAG_PREVIEWCOLORSPACE was added in libtiff 4.5.1. This effectively becomes the minimum required version of libtiff for darktable

> ### What is the version of your OS?
> debian 12

debian 12 has libtiff 4.5.0

--- a/src/imageio/imageio_dng.c
+++ b/src/imageio/imageio_dng.c
@@ -36,6 +36,11 @@
 #include <wchar.h>
 #endif
 
+// fallback for libtiff < 4.6
+#ifndef TIFFTAG_PREVIEWCOLORSPACE
+#define TIFFTAG_PREVIEWCOLORSPACE 50970
+#endif
+
 // DNG uses SRATIONAL / RATIONAL for matrix and WB tags. libtiff accepts
 // these as float/double arrays and handles the conversion; we just pass
 // the values as double

